File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l = 1
        r = max(piles)
        res = r

        while l <= r:
            m = (r + l) // 2
            time = sum([math.ceil(p / m) for p in piles])
            if time <= h:
                res = m
                r = m - 1
            else:
                l = m + 1

        return res

File: Data Structures & Algorithms/max-water-container/submission-2.java
class Solution {
    public int maxArea(int[] heights) {
        int res = 0;
        int left = 0;
        int right = heights.length - 1;

        while (left < right) {
            int total = (right - left) * Math.min(heights[left], heights[right]);
            res = Math.max(total, res);
            if (heights[left] < heights[right]) {
                left++;
            }
            else {
                right--;
            }
        }

        return res;
    }
}
